VimeoNetworking/Sources/Models/VIMUpload.swift

```diff
@@ -0,0 +1,102 @@
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// Encapsulates the upload-related information on a video object.
+public class VIMUpload: VIMModelObject
+{
+    /// The approach for uploading the video, expressed as a Swift-only enum
+    ///
+    /// - streaming: Two-step upload without using tus; this will be deprecated
+    /// - post: Upload with an HTML form or POST
+    /// - pull: Upload from a video file that already exists on the internet
+    /// - tus: Upload using the open-source tus protocol
+    public enum UploadApproach: String
+    {
+        case streaming
+        case post
+        case pull
+        case tus
+    }
+    
+    /// The status code for the availability of the uploaded video, expressed as a Swift-only enum
+    ///
+    /// - complete: The upload is complete
+    /// - error: The upload ended with an error
+    /// - underway: The upload is in progress
+    public enum UploadStatus: String
+    {
+        case complete
+        case error
+        case underway = "in_progress"
+    }
+    
+    /// The approach for uploading the video
+    @objc dynamic public private(set) var approach: String?
+
+    /// The file size in bytes of the uploaded video
+    @objc dynamic public private(set) var size: NSNumber?
+
+    /// The status code for the availability of the uploaded video
+    @objc dynamic public private(set) var status: String?
+    
+    /// The HTML form for uploading a video through the post approach
+    @objc dynamic public private(set) var form: String?
+    
+    /// The link of the video to capture through the pull approach
+    @objc dynamic public private(set) var link: String?
+
+    /// The URI for completing the upload
+    @objc dynamic public private(set) var completeUri: String?
+    
+    /// The redirect URL for the upload app
+    @objc dynamic public private(set) var redirectUrl: String?
+
+    /// The link for sending video file data
+    @objc dynamic public private(set) var uploadLink: String?
+    
+    /// The approach for uploading the video, mapped to a Swift-only enum
+    public private(set) var uploadApproach: UploadApproach?
+    
+    /// The status code for the availability of the uploaded video, mapped to a Swift-only enum
+    public private(set) var uploadStatus: UploadStatus?
+    
+    // MARK: - VIMMappable Protocol Conformance
+    
+    /// Called when automatic object mapping completes
+    public override func didFinishMapping()
+    {
+        if let approachString = self.approach
+        {
+            self.uploadApproach = UploadApproach(rawValue: approachString)
+        }
+        
+        if let statusString = self.status
+        {
+            self.uploadStatus = UploadStatus(rawValue: statusString)
+        }
+    }
+}
```

Comment:
How do you feel about adding a few unit tests to test this model object?

Reply:
I restricted testing to `didFinishMapping` since it's the only place with custom code for mapping.